<launch>
  <!-- Map server -->
  <node pkg="map_server" type="map_server" name="map_server" args="$(find python_astar_dummy_planner)/map/map.yaml"/>

  <!-- AMCL for localization -->
  <node pkg="amcl" type="amcl" name="amcl" output="screen"/>

  <!-- Python A* node -->
  <node pkg="python_astar_dummy_planner" type="simple_astar_planner.py" name="simple_astar_planner" output="screen"/>

  <!-- Static TF odom -> base_link to fake robot pose -->
  <node pkg="tf" type="static_transform_publisher"
        name="fake_odom_tf"
        args="0 0 0 0 0 0 odom base_link 100" />
  
  <!-- Fake static TF: map -> odom -->
  <!-- <node pkg="tf" type="static_transform_publisher"
        name="fake_map_to_odom"
        args="0 0 0 0 0 0 map odom 100" /> -->

  <!-- move_base with your planner -->
  <node pkg="move_base" type="move_base" name="move_base" output="screen">
    <rosparam file="$(find python_astar_dummy_planner)/launch/costmap_common_params.yaml" command="load" />
    <rosparam file="$(find python_astar_dummy_planner)/launch/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find python_astar_dummy_planner)/launch/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find python_astar_dummy_planner)/launch/base_local_planner_params.yaml" command="load" />

    <param name="base_global_planner" value="dummy_global_planner/DummyGlobalPlanner"/>
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>
  </node>
  <!-- RViz with pre-configured view -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find python_astar_dummy_planner)/rviz/view_navigation.rviz" />
</launch>
